io-sim/test/Test/STM.hs

```diff
@@ -676,6 +702,11 @@ genTerm env tyrep =
         OrElse <$> genTerm env tyrep
                <*> genTerm env tyrep
 
+    catchTerm =
+      sized $ \sz -> resize (sz `div` 2) $
```

Comment:
```suggestion
      scale (`div` 2) $
```
(feel free to ignore if you think it is not worth to change `orElseTerm` in the same way)